crypto/bls/attestation.go

```diff
@@ -196,6 +198,11 @@ func (k *KeyPair) SaveToFile(path string, password string) error {
 		return err
 	}
 
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
```

Comment:
Why is this 0755 but the writeFile below is 0644?

Reply:
755 is suited for folder creation 
644 is for non-executable file creation
https://www.maths.cam.ac.uk/computing/linux/unixinfo/perms#:~:text=755%20means%20you%20can%20do,users%20can%20only%20read%20it.

it doesn't matter in our case that much?

Reply:
I see. We might want to err on the more secure side and give 644 then? Why does that directory need to be executable..? Also probably don't want to give "others" the right to change my key file. Maybe 654 could work to give group access to write.. but if they really need that then they can go change it manually?

Reply:
Having execute permission on a directory authorizes you to look at extended information on files in the directory  like you can us `ls -l`

Reply:
We talked offline. Agree 755 for dirs and 644 for files is correct.